src/thumbnailer_compare.cc

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "thumbnailer.h"
+
+namespace libwebp {
+
+Thumbnailer::Status Thumbnailer::AnimData2Pictures(
+    WebPData* const webp_data,
+    std::vector<std::unique_ptr<WebPPicture, void (*)(WebPPicture*)>>* const
+        pics) {
+  std::unique_ptr<WebPAnimDecoder, void (*)(WebPAnimDecoder*)> dec(
+      WebPAnimDecoderNew(webp_data, NULL), WebPAnimDecoderDelete);
+  if (dec == NULL) {
+    std::cerr << "Error parsing image." << std::endl;
+    return kMemoryError;
+  }
+
+  WebPAnimInfo anim_info;
+  if (!WebPAnimDecoderGetInfo(dec.get(), &anim_info)) {
+    std::cerr << "Error getting global info about the animation";
+    return kMemoryError;
```

Comment:
In another PR, rename `kMemoryError` to `kGenericError` or add more `Thumbnailer::Status` because returning a memory error might be misleading here.